src/molecules/Modal/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import ReactModal from 'react-modal';
+import TextComponent from 'atoms/TextComponent';
+import styles from './modal.module.scss';
+
+function Modal(props) {
+  const {
+    children,
+    className,
+    header,
+    isOpen,
+    contentLabel,
+    onAfterOpen,
+    onRequestClose,
+    variant,
+    overlay
+} = props;
+
+  return (
+    <ReactModal
+      isOpen={isOpen}
+      onAfterOpen={onAfterOpen}
+      onRequestClose={onRequestClose}
+      contentLabel={contentLabel}
+      className={classnames(styles['modal'], styles[variant], className)}
+      overlayClassName={styles[overlay]}
+    >
+      <div className={styles.dialog}>
+        <div className={styles.header}>
+          <div
+            className={styles.close}
+            onClick={onRequestClose}
+          />
+          <TextComponent
+            type={2}
+            semibold
+          >
+            {header}
+          </TextComponent>
+        </div>
+
+        <div className={styles.body}>
+          {children}
+        </div>
+      </div>
+    </ReactModal>
+  );
+}
+
+
+Modal.propTypes = {
+  /**
+   * This prop will add a new className to any inherent classNames
+   * provided in the component's index.js file.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Provides header text for modal
+   */
+  header: PropTypes.string,
+
+  /**
+   * label passed to ReactModal `contentLabel`
+   */
+  contentLabel: PropTypes.string,
+
+  /**
+   * Boolean to determine modal open/closed
+   */
+  isOpen: PropTypes.bool,
+
+  /**
+   * callback passed to ReactModal `onAfterOpen`
+   */
+  onAfterOpen: PropTypes.func,
+
+  /**
+   * callback passed to ReactModal `onRequestClose` and to modal 'X'
+   */
+  onRequestClose: PropTypes.func,
+
+  /**
+   * variant for modal
+   */
+  variant: PropTypes.string,
+
+  /**
+   * overlay variant for modal
+   */
+  overlay: PropTypes.string
```

Comment:
I left it as a string in case we want different styles of the overlay. I'm not sure if we should do that anyway